File: Chapter06/bookstore-authorization-server/src/main/java/com/bookstore/auth/user/repository/UserRepository.java
package com.bookstore.auth.user.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.bookstore.auth.user.model.User;

@Repository
public interface UserRepository extends JpaRepository<User, Integer> {
	
	User findByUsername(String username);
	
	User findByEmail(String email);
}

File: chapter07/taskmanagementsystem/src/main/kotlin/com/nilangpatel/tms/repository/UserRepository.kt
package com.nilangpatel.tms.repository

import com.nilangpatel.tms.model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Int> {

    fun findByUsername(username: String): User?
}